posthog/temporal/common/clickhouse.py

```diff
@@ -599,9 +624,32 @@ async def acheck_query(
         elif "QueryStart" in events:
             return ClickHouseQueryStatus.RUNNING
         else:
-            self.logger.warning("Expected event not found in query log", query_id=query_id, events=events)
             raise ClickHouseQueryNotFound(query, query_id)
 
+    async def acheck_query_in_process_list(self, query_id: str) -> bool:
+        """Check if a query is running in the ClickHouse process list.
+
+        Arguments:
+            query_id: The ID of the query to check.
+
+        Returns:
+            True if the query is running, False otherwise.
+        """
+        query = """
+                SELECT 1
+                FROM clusterAllReplicas({{cluster_name:String}}, system.processes)
+                WHERE query_id = {{query_id:String}}
+                    AND NOT is_cancelled
+                LIMIT 1
+                """
+
+        resp = await self.read_query(
+            query,
+            query_parameters={"query_id": query_id, "cluster_name": settings.CLICKHOUSE_CLUSTER},
+            query_id=f"{query_id}-CHECK-PROCESS-LIST",
+        )
+        return bool(resp)
```

Comment:
hmm might be worth checking that resp is actually 1. When no rows match, the response will be empty, and if that means anything besides empty string, for example something like `bool("\n")`, then that will evaluate to `True`.

Reply:
yeah, I saw the test and that's why I approved the PR, but I was still a bit scared...

Reply:
have updated that now